source/common/upstream/upstream_impl.h

```diff
@@ -301,7 +303,7 @@ class HostDescriptionImpl : virtual public HostDescription,
   const std::string health_checks_hostname_;
   Network::Address::InstanceConstSharedPtr address_;
   // The first entry in the address_list_ should match the value in address_.
-  std::vector<Network::Address::InstanceConstSharedPtr> address_list_;
+  std::shared_ptr<std::vector<Network::Address::InstanceConstSharedPtr>> address_list_;
```

Comment:
per comments & chat I think we should mostly leave the base class impl alone, and put a mutex-guarded `shared_ptr<vector>` in the derived class where it is needed.

If the base-class member var is still actually read we can take one of a few approaches to make it unique. I'm not sure it really is (other than init). But if it is we can rename it and use a virtual accessor to read it, and the subclass can override. But it'd be better to confirm.